src/ray/gcs/gcs_server/test/gcs_task_manager_test.cc

```diff
@@ -813,11 +740,12 @@ TEST_F(GcsTaskManagerMemoryLimitedTest, TestLimitReturnRecentTasksWhenGetAll) {
     // Add a task event
     {
       inserted++;
-      auto events = GenTaskEvents({task_ids[i]},
-                                  /* attempt_number */ 0,
-                                  /* job_id */ 0,
-                                  /* profile event */ absl::nullopt,
-                                  GenStateUpdate({{rpc::TaskStatus::RUNNING, 1}}));
+      auto events =
```

Comment:
Can you actually add 2 more tests here (if not covered)?

1. Make sure MarkTaskFailed series (job, task, and worker) is idempotent (i,e., it doesn't overwriting failed_ts)
2. the internal indexes are properly cleaned up. Something similar to https://github.com/ray-project/ray/blob/1f7058ee40b32c49b353e8e00806325744122d17/src/ray/pubsub/test/subscriber_test.cc#L407

Reply:
1. It's tested in https://github.com/ray-project/ray/blob/1e21e1a8e8c7c45dae3e592bd38f1baf2c860aff/src/ray/gcs/gcs_server/test/gcs_task_manager_test.cc#L495

Added another unit test on the function specifically. 

3. is covered in https://github.com/ray-project/ray/blob/1e21e1a8e8c7c45dae3e592bd38f1baf2c860aff/src/ray/gcs/gcs_server/test/gcs_task_manager_test.cc#L568